pennylane/devices/tests/test_gates.py

```diff
@@ -603,3 +603,333 @@ def circuit():
         mat = mat.conj().T
         expected = np.abs(mat @ rnd_state) ** 2
         assert np.allclose(res, expected, atol=tol(dev.shots))
+
+
+@flaky(max_runs=10)
+class TestGatesQubitExpval:
```

Comment:
All of these tests were adopted from the tests for `DefaultQubit` (which are later removed in this PR).

The added value in having these tests is that we now have cases which test gate applications with `qml.expval`.

Reply:
Nice!

While potentially redundant at the moment, since `QubitDevice` uses probabilities + eigenvalues to compute expectation values, this is definitely worth having, since we could have devices that use separate logic for expectation values.

Reply:
Do you think it's worth having this in a separate file? This file is becoming quite large.

Reply:
Definitely, that's a great point :) moved them to a new test in d8cb74fd931c13831cbae8ab812fe35b7e894641